### PR TITLE
separate online and offline transitions in metrics

### LINKED ISSUE #893: create or derive metric for number of online interaction transitions
NOT including demonstration transitions (which are included in "NUM_TRANSITIONS").

### PR DESCRIPTION
**Pull request by tomsilver**
closes #893 

@amburger66 this change is going to break the plotting script for any results that were previously generated. to plot old results, we could modify the script locally as needed. or, we can make this change backward compatible, but i'm worried then that we're going to get confused about what "num transitions" means... let me know what you think

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -147,7 +149,7 @@ def _run_pipeline(env: BaseEnv,
                 continue
             logging.info(f"\n\nONLINE LEARNING CYCLE {i}\n")
             logging.info("Getting interaction requests...")
-            if total_num_transitions > CFG.online_learning_max_transitions:
+            if num_online_transitions > CFG.online_learning_max_transitions:
```
Comment (tomsilver) [inline src/main.py:152]: @amburger66 note: we're capping _online_ transitions now, so we're going to be running the scripts for longer. I think this is more intuitive?

File: scripts/plotting/create_num_demos_plots.py [hunk 2]
```diff
@@ -135,7 +134,7 @@ def _main() -> None:
                                 label=label,
                                 marker=marker)
                 # Automatically make x ticks integers for certain X KEYS.
-                if x_key in ("CYCLE", "NUM_TRANSITIONS"):
+                if x_key in ("CYCLE", "NUM_OFFLINE_TRANSITIONS"):
```
Comment (ronuchit) [inline scripts/plotting/create_num_demos_plots.py:137]: will this script still work correctly? it looks like you didn't add `NUM_OFFLINE_TRANSITIONS` to `X_KEY_AND_LABEL` above

Comment (tomsilver) [reply scripts/plotting/create_num_demos_plots.py:138]: fixed

File: src/main.py (hunk 1 above)
Comment (amburger66) [reply src/main.py]: makes sense

**Review by amburger66 — APPROVED** [review]

**Comment by amburger66** [comment]
> @amburger66 this change is going to break the plotting script for any results that were previously generated. to plot old results, we could modify the script locally as needed. or, we can make this change backward compatible, but i'm worried then that we're going to get confused about what "num transitions" means... let me know what you think

i think this is fine. thanks!

**Pull request merged**